gensim/utils.py

```diff
@@ -1278,3 +1277,456 @@ def lazy_flatten(nested_list):
                 yield sub
         else:
             yield el
+
+
+class PorterStemmer(object):
```

Comment:
I think a similar comment applies here to my comment above about the `coherence_utils` module. I think this would be better organized as a package, with subpackages for `preprocessing`, `corpora`, etc. I think the existing `utils` module was already getting a bit bloated. I would hypothesize very long files with a wide variety of things makes it hard to find things for folks who are new to the codebase. I might even advocate a more granular decomposition such as: `utils/stemming`, `utils/lemmatizing`, `utils/corpora`, `utils/helpers`, or something of that sort, with aggregation in the `utils/__init__`.

Reply:
In general, I agree (maybe not very "detailed" utils, but splitting `gensim.utils` to `gensim.utils.utils` (old) and `gensim.utils.preprocessing` is better)